src/libsyntax/ast.rs

```diff
@@ -1746,13 +1730,21 @@ impl StructFieldKind {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+pub enum VariantKind {
+    Dict,
```

Comment:
Is there a different name than `Dict` for these (I thought about "struct-like variant", but that was too confusing). There are some differences here - e.g. tuple-like variants can't be used in FRU. There's also the thing that `struct S {}` does not have a value namespace entry, but I don't remember if these are `VariantKind::Unit` or `VariantKind::Dict`.